src/sentry/api/serializers/models/alert_rule.py

```diff
@@ -148,17 +148,16 @@ def get_attrs(
 
         alert_rule_projects = set()
         for alert_rule in alert_rules.values():
-            try:
-                alert_rule_projects.add([alert_rule.id, alert_rule.projects.get().slug])
-            except Exception:
-                pass
+            if alert_rule.projects.exists():
+                for project in alert_rule.projects.all():
+                    alert_rule_projects.add((alert_rule.id, project.slug))
 
         # TODO - Cleanup Subscription Project Mapping
         snuba_alert_rule_projects = AlertRule.objects.filter(
             id__in=[item.id for item in item_list]
         ).values_list("id", "snuba_query__subscriptions__project__slug")
 
-        alert_rule_projects.update(snuba_alert_rule_projects)
+        alert_rule_projects.update([tup for tup in snuba_alert_rule_projects if tup[1]])
```

Comment:
wonder if t here's a better variable than `tup` to represent `tuple`... 🤔 maybe for readability we could do like:
```
[(rule_id, slug) for rule_id, slug in snuba_alert_rule_projects if slug]
```
Feels a little verbose compared to what you have though. It just took me a second to grok this line cause of the short variable name with the keywords and stuff.